controlplane/kubeadm/internal/workload_cluster_rbac.go

```diff
@@ -66,6 +74,31 @@ func (w *Workload) EnsureResource(ctx context.Context, obj client.Object) error
 	return nil
 }
 
+// AddClusterAdminRoleBinding creates ClusterRoleBinding rules to use the kubeadm:cluster-admins Cluster Role created in Kubeadm v1.29.
+func (w *Workload) AddClusterAdminRoleBinding(ctx context.Context, version semver.Version) error {
+	// If the upgrade is not to a Cluster of version v1.29.0 or higher this is a no-op
+	if !version.GTE(semver.MustParse("1.29.0")) {
```

Comment:
if you are running this code in CI against kubeadm CI artifacts, the exact version where this feature was somehow useable and the RBAC was needed is:
v1.29.0-alpha.2.188+05076de57fc49f

Reply:
Updated to use the CAPI util that ignores preRelease versions for checks like these. Thanks for catching this!